ui/builder.js

```diff
@@ -0,0 +1,188 @@
+const fs = require('fs-extra')
+const os = require('os')
+const { start } = require('live-server')
+const { createProxyMiddleware } = require('http-proxy-middleware')
+const { watch } = require('chokidar')
+const { build } = require('esbuild')
+// TODO: submit PR for upstream
+fs.copyFileSync(
+  './postcss2-index.js',
+  './node_modules/esbuild-plugin-postcss2/dist/index.js'
+)
```

Comment:
We hack the esbuild-plugin-postcss2 by fixing some bugs, adding cache, we need to feedback to the upstream later.